File: internal/model/game_team.go
package model

type GameTeam struct {
	ID        uint   `json:"id,omitempty"`
	TeamID    uint   `gorm:"uniqueIndex:game_team_idx" json:"team_id,omitempty"`
	Team      *Team  `json:"team,omitempty"`
	GameID    uint   `gorm:"uniqueIndex:game_team_idx" json:"game_id,omitempty"`
	Game      *Game  `json:"game,omitempty"`
	IsAllowed *bool  `gorm:"default:false;not null;" json:"is_allowed,omitempty"`
	Signature string `gorm:"unique" json:"signature,omitempty"`
}

File: web/src/types/game_team.ts
import { Game } from "./game";
import { Team } from "./team";

export interface GameTeam {
    id?: number;
    team_id?: number;
    team?: Team;
    game_id?: number;
    game?: Game;
    rank?: number;
    pts?: number;
    solved?: number;
    is_allowed?: boolean;
    signature?: string;
}
